MobilityData/mobility-feed-api
# feat: set feature flag system in place

**Summary:**  
Closes #329 
This PR integrates a feature flag system by leveraging Firebase Remote Config capabilities. 

To add a new feature flag, simply append its name and default value to `interface/RemoteConfig.ts`. The Firebase console allows for the remote modification of these flags, overriding their default values as soon as the cache expires.

**Expected Behavior:**  
Feature flags serve as switches for enabling or disabling application features. The implementation facilitates server-side updates and control over feature availability directly from the Firebase console.

**Note about update rate of feature flags:**  
In the production environment, the feature flag update rate is configured to 12 hours, adhering to Firebase's best practice recommendations to prevent throttling. For development and QA environments, this rate is reduced to 5 minutes, allowing for rapid testing and iteration. More details on throttling can be found in the [Firebase documentation](https://firebase.google.com/docs/remote-config/get-started?hl=en&authuser=0&_gl=1*jbxp92*_ga*NjEwMjk3NjY5LjE3MTAzNjIzMTg.*_ga_CW55HF8NVT*MTcxMDM2MjMxOC4xLjEuMTcxMDM2MjMyNC41NS4wLjA.&platform=web#throttling).

**Testing Instructions:**  
To test the functionality of a feature flag:
1. Navigate to the Firebase Console.
2. Access the `mobility-feeds-dev` project.
3. Select Build > Remote Config from the menu.
4. Locate the `enableGithubSSO` configuration and set its value to `false`.
5. Allow up to 5 minutes for the cache to update.
6. Upon visiting the login or sign-up page, the option to use GitHub for authentication should be disabled.


Please make sure these boxes are checked before submitting your pull request - thanks!
